### PR TITLE
feat(Sunsetting-sesocio): signup-latam

### PR DESCRIPTION
**Pull request by fgonzalez-bc**
Minnor changes in order to show a custom wallet signup for Latam user coming from sesocio.com

image

### CODE REVIEW

**Review by cmonti-bc — APPROVED** [review]

File: packages/blockchain-wallet-v4-frontend/src/components/Terms/index.tsx [hunk 1]
```diff
@@ -80,7 +80,7 @@ const Terms = (props) => {
             data-e2e='blockchainTermsLink'
           >
             <FormattedMessage
-              id='scenes.register.registerform.blockchain.default.terms'
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/components/Terms/index.tsx:83]: why did you change this line?

File: packages/blockchain-wallet-v4-frontend/src/layouts/Public/components/Header/index.tsx [hunk 2]
```diff
@@ -47,7 +50,7 @@ const Header = (props) => {
         <NavbarBrandStyled>
           <PublicBrand>
             <HeaderLink href='https://www.blockchain.com'>
-              <BlockchainLogoImage name='blockchain-logo' height='24px' />
+            {!isLatam ? <BlockchainLogoImage name='blockchain-logo' height='24px' /> : <BlockchainLogoImage name='sesocio-to-blockchain-logo' height='70px' /> }
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/layouts/Public/components/Header/index.tsx:53]: could you inverse this check? it makes it hard to read like this

File: packages/blockchain-wallet-v4-frontend/src/scenes/Signup/components/Header/index.tsx [hunk 3]
```diff
@@ -36,20 +38,31 @@ const SubHeader = styled(Text)`
...
-          defaultMessage='Securely Buy, Sell, and Store Crypto.'
-        />
+        {!props.isLatam && (
+          <FormattedMessage
+            id='scenes.register.securelybuy'
+            defaultMessage='Securely Buy, Sell, and Store Crypto.'
+          />
+        )}
+        {props.isLatam && (
+          `Envía, recibe y obtén hasta un 14g% de ganancias con tus criptos.`
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Signup/components/Header/index.tsx:52]: why is this not translated?

File: packages/blockchain-wallet-v4-frontend/src/scenes/Signup/components/Header/index.tsx [hunk 4]
```diff
@@ -36,20 +38,31 @@ const SubHeader = styled(Text)`
...
+      
       <SubHeader color='whiteFade800' weight={500} data-e2e='signupSecureSubHeader'>
+      {!props.isLatam && (
         <FormattedMessage
           id='scenes.register.getstarted'
           defaultMessage='Get Started by Signing Up Now.'
         />
+      )}
+      {props.isLatam && (
+        `Crea tu cuenta y verifica tu perfil para comenzar a operar.`
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Signup/components/Header/index.tsx:64]: this needs translation

File: packages/blockchain-wallet-v4-frontend/src/scenes/Signup/SignupLanding/index.tsx [hunk 5]
```diff
@@ -45,12 +45,19 @@ const ExchangeCardWrapper = styled.div<{ $showForm: boolean }>`
...
+`
+
 const SignupLanding = (props: InjectedFormProps<{}> & SubviewProps) => (
   <>
-    <Header />
+    <Header {...props} />
     <CardsWrapper>
       <SignupCard {...props} />
-
+      {!props.isLatam && (
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Signup/SignupLanding/index.tsx:60]: can you use  ? : here?

File: packages/blockchain-wallet-v4-frontend/src/layouts/Public/index.tsx [hunk 6]
```diff
@@ -83,7 +86,7 @@ const PublicLayoutContainer = ({
             <Alerts />
 
             <HeaderContainer>
-              <Header authProduct={authProduct} />
+              <Header authProduct={authProduct} urlParams={matchProps.location} />
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/layouts/Public/index.tsx:89]: why are you passing this prop here?

File: packages/blockchain-wallet-v4-frontend/src/layouts/Public/index.tsx [hunk 7]
```diff
@@ -26,10 +26,12 @@ const FooterContainer = styled.div`
     margin-top: 8px;
   `}
 `
+const QsParams = new URLSearchParams(window.location.hash)
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/layouts/Public/index.tsx:29]: can this be named **qsParams**?

File: packages/blockchain-wallet-v4-frontend/src/layouts/Public/components/Header/index.tsx [hunk 8]
```diff
@@ -6,12 +6,15 @@ import Announcements from 'components/Announcements'
 import { Brand, Public } from 'components/NavbarV2'
 import { media } from 'services/styles'
 
+const QsParams = new URLSearchParams(window.location.hash)
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/layouts/Public/components/Header/index.tsx:9]: can this be named **qsParams**?

File: packages/blockchain-wallet-v4-frontend/src/layouts/Public/components/Header/index.tsx [hunk 9]
```diff
@@ -6,12 +6,15 @@ import Announcements from 'components/Announcements'
...
+const isLatam = QsParams.has('latam')
+
 const NavbarStyled = styled(Public)<{ authProduct: string }>`
   padding: 0 16px;
   box-sizing: border-box;
   background-color: ${(props) =>
-    props.authProduct === 'EXCHANGE' ? props.theme.exchangeLogin : props.theme.grey900};
-  background-image: ${(props) => props.authProduct !== 'EXCHANGE' && `url('/img/bg-pattern.svg')`};
+    props.authProduct === 'EXCHANGE' ? props.theme.exchangeLogin : (isLatam) ? '#020109' : props.theme.grey900};
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/layouts/Public/components/Header/index.tsx:16]: is there a way to prevent this nested ternary?

**Review by pedroapfilho — CHANGES_REQUESTED** [review]

File: packages/blockchain-wallet-v4-frontend/src/components/Terms/index.tsx (hunk 1 above)
Comment (fgonzalez-bc) [reply packages/blockchain-wallet-v4-frontend/src/components/Terms/index.tsx:83]: because it seems there is a bug in production ? take a look to the screenshot
image

File: packages/blockchain-wallet-v4-frontend/src/scenes/Signup/components/Header/index.tsx (hunk 3 above)
Comment (fgonzalez-bc) [reply packages/blockchain-wallet-v4-frontend/src/scenes/Signup/components/Header/index.tsx:52]: I am waiting to have access to lokalize. But this latam signup page will be in production on April 18 is everything goes ok.

File: packages/blockchain-wallet-v4-frontend/src/scenes/Signup/components/Header/index.tsx (hunk 4 above)
Comment (fgonzalez-bc) [reply packages/blockchain-wallet-v4-frontend/src/scenes/Signup/components/Header/index.tsx:64]: I am waiting to have access to lokalize app.

File: packages/blockchain-wallet-v4-frontend/src/scenes/Signup/SignupLanding/index.tsx (hunk 5 above)
Comment (fgonzalez-bc) [reply packages/blockchain-wallet-v4-frontend/src/scenes/Signup/SignupLanding/index.tsx:60]: Yes!

File: packages/blockchain-wallet-v4-frontend/src/layouts/Public/index.tsx (hunk 6 above)
Comment (fgonzalez-bc) [reply packages/blockchain-wallet-v4-frontend/src/layouts/Public/index.tsx:89]: removed!

File: packages/blockchain-wallet-v4-frontend/src/layouts/Public/index.tsx (hunk 7 above)
Comment (fgonzalez-bc) [reply packages/blockchain-wallet-v4-frontend/src/layouts/Public/index.tsx:29]: yes!

File: packages/blockchain-wallet-v4-frontend/src/layouts/Public/components/Header/index.tsx (hunk 8 above)
Comment (fgonzalez-bc) [reply packages/blockchain-wallet-v4-frontend/src/layouts/Public/components/Header/index.tsx:9]: yes!

File: packages/blockchain-wallet-v4-frontend/src/layouts/Public/components/Header/index.tsx (hunk 9 above)
Comment (fgonzalez-bc) [reply packages/blockchain-wallet-v4-frontend/src/layouts/Public/components/Header/index.tsx:16]: That wasn't me, I just added an extra condition.

File: packages/blockchain-wallet-v4-frontend/src/scenes/Signup/components/Header/index.tsx (hunk 3 above)
Comment (pedroapfilho) [reply packages/blockchain-wallet-v4-frontend/src/scenes/Signup/components/Header/index.tsx]: @schnogz could you give him access to lokalize?

File: packages/blockchain-wallet-v4-frontend/src/scenes/Signup/components/Header/index.tsx [hunk 10]
```diff
@@ -36,20 +38,37 @@ const SubHeader = styled(Text)`
...
+            id='scenes.register.securelybuy'
+            defaultMessage='Securely Buy, Sell, and Store Crypto.'
+          />
+        )}
+        {props.isLatam && (
+          <FormattedMessage
+            id='scenes.register.sendreceiveearn'
+            defaultMessage='Send, receive and get up to 14% profit with your cryptos.'
+          />
+        )}
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Signup/components/Header/index.tsx:56]:
```suggestion
        {props.isLatam ?
        (
         <FormattedMessage
            id='scenes.register.sendreceiveearn'
            defaultMessage='Send, receive and get up to 14% profit with your cryptos.'
          />
        )
        :
         (
          <FormattedMessage
            id='scenes.register.securelybuy'
            defaultMessage='Securely Buy, Sell, and Store Crypto.'
          />
        )}
```

**Review by pedroapfilho — CHANGES_REQUESTED** [review]

**Review by pedroapfilho — APPROVED** [review]

File: packages/blockchain-wallet-v4-frontend/src/layouts/Public/components/Header/index.tsx [hunk 11]
```diff
@@ -6,12 +6,16 @@ import Announcements from 'components/Announcements'
...
+const qsParams = new URLSearchParams(window.location.hash)
+const isLatam = qsParams.has('latam')
+
 const NavbarStyled = styled(Public)<{ authProduct: string }>`
   padding: 0 16px;
   box-sizing: border-box;
   background-color: ${(props) =>
-    props.authProduct === 'EXCHANGE' ? props.theme.exchangeLogin : props.theme.grey900};
-  background-image: ${(props) => props.authProduct !== 'EXCHANGE' && `url('/img/bg-pattern.svg')`};
+    props.authProduct === 'EXCHANGE' ? props.theme.exchangeLogin : (isLatam) ? '#020109' : props.theme.grey900};
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/layouts/Public/components/Header/index.tsx:16]: why not using color from theme?

File: packages/blockchain-wallet-v4-frontend/src/scenes/Signup/SignupLanding/index.tsx [hunk 12]
```diff
@@ -45,12 +45,19 @@ const ExchangeCardWrapper = styled.div<{ $showForm: boolean }>`
...
+  display: flex;
+  flex-direction: column;
+  margin-left: 100px;
+  color: white;
+`
+
 const SignupLanding = (props: InjectedFormProps<{}> & SubviewProps) => (
   <>
-    <Header />
+    <Header {...props} />
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Signup/SignupLanding/index.tsx:57]: we should try to stop spreading all props, and just necessary ones

**Review by milan-bc — APPROVED** [review]
